website/docs/metaprogramming-plugins.md

```diff
@@ -0,0 +1,124 @@
+---
+id: metaprogramming-plugins
+title: Metaprogramming plugins
+---
+
+> **Note**: The plugin system has many drawbacks and is not considered a stable
+> feature. Expect its API to change in the future.
+
+In Ruby, it is possible to dynamically define methods. Consider the following
+example:
+
+```ruby
+# metaprogramming.rb
+# typed: true
+
+class MetaProgramming
```

Comment:
Can we consistently use `Metaprogramming` ? This is how the wikipedia article stylizes it.